src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -69,31 +70,73 @@ private async Task<string> GetRegionAsync(ICoreLogger logger)
 
             try
             {
-                HttpResponse response = await _httpManager.SendGetAsync(_ImdsUri, Headers, logger).ConfigureAwait(false);
+                var headers = new Dictionary<string, string>
+                {
+                    { "Metadata", "true" }
+                };
+                
+                HttpResponse response = await _httpManager.SendGetAsync(BuildImdsUri(DefaultApiVersion), headers, logger).ConfigureAwait(false);
```

Comment:
As discussed, I believe you want to make some changes here - no retry + timeout. Not sure if part of the same PR or not.

Reply:
It is not a part of this PR. I created an issue to track those changes: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2177